src/Extensions/Activities/Internal/EventSource/ActivityEventSender.cs

```diff
@@ -36,9 +36,29 @@ public void SendActivityMetric(Activity activity)
 			string userHash = activity.GetUserHash();
 			bool isHealthCheck = activity.IsHealthCheck();
 
-			string subtype = activity.GetBaggageItem(ActivityTagKeys.SubType) ?? NullPlaceholder;
-			string metadata = activity.GetBaggageItem(ActivityTagKeys.Metadata) ?? NullPlaceholder;
-			string resultAsString = activity.GetBaggageItem(ActivityTagKeys.Result) ?? NullPlaceholder;
+			string subtype = NullPlaceholder;
+			string metadata = NullPlaceholder;
+			string resultAsString = NullPlaceholder;
+			foreach (KeyValuePair<string, string?> pair in activity.Tags)
+			{
+				if (pair.Value == null)
+				{
+					continue;
+				}
+
+				if (string.Equals(ActivityTagKeys.Result, pair.Key, StringComparison.Ordinal))
+				{
+					resultAsString = pair.Value;
+				}
+				else if (string.Equals(ActivityTagKeys.SubType, pair.Key, StringComparison.Ordinal))
+				{
+					subtype = pair.Value;
+				}
+				else if (string.Equals(ActivityTagKeys.Metadata, pair.Key, StringComparison.Ordinal))
+				{
+					metadata = pair.Value;
+				}
```

Comment:
```suggestion
				switch (pair.Key)
				{
					case ActivityTagKeys.Result:
						resultAsString = pair.Value;
					case ActivityTagKeys.SubType:
						subtype = pair.Value;
					case ActivityTagKeys.Metadata:
						metadata = pair.Value;
					default:
						break;					
				}

Reply:
Thanks for the suggestion it definitely makes the code look better, but all changes in this file are reverted now and merging it will create conflict.  
I would like to avoid changing this file. My hope is that we can remove `ActivityEventSender.cs` eventually. Also using switch replaces string compare with regular `Equals`. I don't remember the details but I think `string.Equals(..., ..., StringComparison.Ordinal)` works faster when you compare only ASCII symbols.